Add WAS table schema

cduhn17 opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

Adding table to database schema: `was_tracker_customerdata`

## 💭 Motivation and context ##

The WAS team will use P&E database instance to house the data for reports that are run by the WAS team. Additionally, this table will be available via P&E UI and REST API endpoint. 


## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->
- [x] The table schema has been added to data_schema.sql
- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

Commit:
Add WAS table schema

cduhn17 committed:
Update was_tracker_customerdata to all text fields

cduhn17 committed:
Merge branch 'develop' into CD-pe-reports-add-WAS-schema

dav3r commented on src/pe_reports/data/data_schema.sql:
```diff
@@ -558,6 +558,35 @@ CREATE TABLE public.unique_software (
+
+create table was_tracker_customerdata
+(
+    customer_id              uuid default uuid_generate_v1() not null
+        constraint was_tracker_customerdata_pk
+            primary key,
+    tag                      text                            not null,
+    customer_name            text                            not null,
+    testing_sector           text                            not null,
+    ci_type                  text                            not null,
+    ticket                   text                            not null,
+    next_scheduled           text                            not null,
+    last_scanned             text                            not null,
+    frequency                text                            not null,
+    comments_notes           text                            not null,
+    was_report_poc           text                            not null,
+    was_report_email         text                            not null,
+    onboarding_date          text                            not null,
+    no_of_web_apps           integer                         not null,
+    no_web_apps_last_updated text,
+    elections                text
```
Can these fields be alphabetized or does their order matter here?  Alphabetical ordering of long lists like this is always preferable for readability, as long as the order does not matter.

cduhn17 replied:
@dav3r , 
The WAS data isn't really mixing with our data at any level at this point and no real talk of integration with their data. We are just providing a space for their data and API endpoints for their team members interact with their data. I left the columns in the order that they had them so I dont change their process of writing to their flat files as they have code in place to write the flat file in this format. I do agree with you the columns s/b in alphabetical order, just doesn't fit the current process.

dav3r replied:
If you are just giving them a place to put their data and there are no plans to integrate it with P&E data, is there any reason why they are not getting their own database, as opposed to adding a (essentially unrelated) WAS table in the P&E database?

cduhn17 replied:
I should rephrase that, our data isn't twisted with theirs, but we will be pulling the data to report on the data in our 
monthly reports.

dav3r replied:
Got it- so it sounds like there is a need to have WAS data in the P&E database.  Thanks for clarifying!

dav3r commented on src/pe_reports/data/data_schema.sql:
```diff
@@ -558,6 +558,35 @@ CREATE TABLE public.unique_software (
+
+create table was_tracker_customerdata
+(
+    customer_id              uuid default uuid_generate_v1() not null
+        constraint was_tracker_customerdata_pk
+            primary key,
+    tag                      text                            not null,
+    customer_name            text                            not null,
+    testing_sector           text                            not null,
+    ci_type                  text                            not null,
+    ticket                   text                            not null,
+    next_scheduled           text                            not null,
+    last_scanned             text                            not null,
+    frequency                text                            not null,
+    comments_notes           text                            not null,
+    was_report_poc           text                            not null,
+    was_report_email         text                            not null,
+    onboarding_date          text                            not null,
+    no_of_web_apps           integer                         not null,
+    no_web_apps_last_updated text,
+    elections                text
```
Looks like you are using different spacing and capitalization (e.g. `not null` vs. `NOT NULL`) than the rest of this file.  These things should be consistent throughout the file.

cduhn17 replied:
@dav3r , 
I have updated to capitalization per commit [c8befc4](https://github.com/cisagov/pe-reports/pull/427/commits/c8befc498db0e2d859499ea0a636994f534b8e3d)

dav3r replied:
What about the spacing differences here vs. the rest of the file?

cduhn17 replied:
@dav3r , 
I have addressed your comment in commit  [1316325](https://github.com/cisagov/pe-reports/pull/427/commits/13163255da333acc357b99c8bc48169fb228c316)

dav3r reviewed (commented):
I left a couple of comments for your consideration.

More generally, I just wanted to confirm that it makes sense to include this WAS data in the P&E database.  If you are ok tying their data to yours (and all of the access and maintenance ramifications that come with that), then it's fine.  I am only making sure that you have thought about how this might affect things a year or two (or longer!) down the road.

Commit:
Upate not null to capitalized

Commit:
Update package versions

dav3r commented on setup.py:
```diff
@@ -122,13 +122,13 @@ def get_version(version_file):
         "pymongo == 4.0.1",
-        "pymupdf == 1.19.0",
+        "pymupdf",
         "python-dateutil >= 2.7.3",
         "pytest-cov",
         "python-pptx == 0.6.21",
         "pytz",
         "pyyaml == 6.0",
-        "reportlab == 3.6.6",
+        "reportlab",
```
The PR description should mention that you are removing these version pins and why you are doing so.

cduhn17 replied:
I have reverted these changes. It originally looked as though these pinned packages were causing problems.

dav3r replied:
Reverted in https://github.com/cisagov/pe-reports/pull/427/commits/1c8b9b49bf1d187d28773c186d4dd4c91aa59b87.

Commit:
dav3r
dav3r reviewed 3 hours ago
setup.py
        "pymupdf",
                "python-dateutil >= 2.7.3",
                        "pytest-cov",
                                "python-pptx == 0.6.21",
                                        "pytz",
                                                "pyyaml == 6.0",
                                                        "reportlab == 3.6.6",
                                                                "reportlab",
                                                                Member
                                                                @dav3r dav3r 3 hours ago
                                                                The PR description should mention that you are removing these version pins and why you are doing so.

                                                                @cduhn17    Reply...
                                                                Merge state
                                                                Add more commits by pushing to the CD-pe-reports-add-WAS-schema branch on cisagov/pe-reports.

                                                                Code owner review required
                                                                Waiting on code owner review from DJensen94, aloftus23, dav3r, felddy, jsf9k, mcdonnnj, and/or stewartl97.
                                                                6 pending reviewers
                                                                View
                                                                Unresolved conversations
                                                                3 conversations must be resolved before merging.
                                                                Some checks were not successful
                                                                10 failing, 3 successful, 4 skipped, and 4 expected checks
                                                                @github-actions
                                                                build / lint (pull_request) Failing after 2m
                                                                Required
                                                                Details
                                                                @github-actions
                                                                build / lint (push) Failing after 2m
                                                                Required
                                                                Details
                                                                @github-actions
                                                                CodeQL / Analyze (python) (pull_request) Successful in 22m
                                                                Required
                                                                Details
                                                                @github-actions
                                                                CodeQL / Analyze (python) (push) Successful in 22m
                                                                Required
                                                                Details
                                                                @github-actions
                                                                build / test (3.7) (pull_request) Failing after 48s
                                                                Required
                                                                Details
                                                                @github-actions
                                                                build / test (3.7) (push) Failing after 1m
                                                                Required
                                                                Details
                                                                @github-actions
                                                                build / test (3.8) (pull_request) Failing after 48s
                                                                Required
                                                                Details
                                                                @github-actions
                                                                build / test (3.8) (push) Failing after 49s
                                                                Required
                                                                Details
                                                                @github-actions
                                                                build / test (3.9) (pull_request) Failing after 49s
                                                                Required
                                                                Details
                                                                @github-actions
                                                                build / test (3.9) (push) Failing after 55s
                                                                Required
                                                                Details
                                                                @github-actions
                                                                build / test (3.10) (pull_request) Failing after 2m
                                                                Required
                                                                Details
                                                                @github-actions
                                                                build / test (3.10) (push) Failing after 2m
                                                                Required
                                                                Details
                                                                @github-actions
                                                                build / coveralls-finish (pull_request) Skipped
                                                                Required
                                                                Details
                                                                @github-actions
                                                                build / coveralls-finish (push) Skipped
                                                                Required
                                                                Details
                                                                @github-actions
                                                                build / build (pull_request) Skipped
                                                                Details
                                                                @github-actions
                                                                build / build (push) Skipped
                                                                Details
                                                                build (3.10) Expected — Waiting for status to be reported
                                                                Required
                                                                build (3.7) Expected — Waiting for status to be reported
                                                                Required
                                                                build (3.8) Expected — Waiting for status to be reported
                                                                Required
                                                                build (3.9) Expected — Waiting for status to be reported
                                                                Required
                                                                @github-code-scanning
                                                                Code scanning results / CodeQL Successful in 4s — No new or fixed alerts
                                                                Required
                                                                Details
                                                                Merging is blocked
                                                                Merging can be performed automatically with 2 approving reviews.
                                                                 You can also open this in GitHub Desktop or view .
                                                                 @cduhn17

                                                                  Add heading textAdd bold text, <Cmd+b>Add italic text, <Cmd+i>
                                                                  Add a quote, <Cmd+Shift+.>Add code, <Cmd+e>Add a link, <Cmd+k>
                                                                  Add a bulleted list, <Cmd+Shift+8>Add a numbered list, <Cmd+Shift+7>Add a task list, <Cmd+Shift+l>
                                                                  Directly mention a user or team
                                                                  Reference an issue, pull request, or discussion
                                                                  Add saved reply
                                                                  Leave a comment
                                                                  No file chosen
                                                                  Attach files by dragging & dropping, selecting or pasting them.
                                                                  Styling with Markdown is supported
                                                                  Remember, contributions to this repository should follow its contributing guidelines and security policy.
                                                                   ProTip! Add .patch or .diff to the end of URLs for Git’s plaintext views.
                                                                   Reviewers
                                                                   @dav3r
                                                                   dav3r
                                                                   dav3r left review comments
                                                                   Re-request review
                                                                   @aloftus23
                                                                   aloftus23
                                                                   @DJensen94
                                                                   DJensen94
                                                                   @felddy
                                                                   felddy
                                                                   @jsf9k
                                                                   jsf9k
                                                                   @mcdonnnj
                                                                   mcdonnnj
                                                                   @stewartl97
                                                                   stewartl97
                                                                   At least 2 approving reviews are required to merge this pull request.

                                                                   Still in progress?
                                                                   Assignees
                                                                   @cduhn17
                                                                   cduhn17
                                                                   Labels
                                                                   Draft PR
                                                                   PR in draft mode, not passing checks
                                                                   Projects
                                                                   Sprint 20
                                                                   PR
                                                                   Milestone
                                                                   No milestone
                                                                   Development
                                                                   Successfully merging this pull request may close these issues.

                                                                   None yet

                                                                   Notifications
                                                                   Customize
                                                                   You’re receiving notifications because you were assigned.
                                                                   3 participants
                                                                   @cduhn17
                                                                   @coveralls
                                                                   @dav3r
                                                                   Revert "Update package versions"

This reverts commit 7253164e3d65963e4e0c952e0ba8b357d0f34453.

cduhn17 committed:
Merge branch 'develop' into CD-pe-reports-add-WAS-schema

aloftus23 reviewed (approved):
Looks good

Commit:
Update spacing at field descriptions

dav3r commented on src/pe_reports/data/data_schema.sql:
```diff
@@ -558,6 +558,33 @@ CREATE TABLE public.unique_software (
 );
 
+--
+--Name: was_tracker_customerdata
```
This should match the other tables:
```suggestion
-- Name: was_tracker_customerdata; Type: TABLE; Schema: public; Owner: pe
```

dav3r reviewed (commented):
The PR description also mentions `vw_cidrs`, but I don't see anything in this PR about a table or schema with that name.

cduhn17 committed:
Update src/pe_reports/data/data_schema.sql

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

cduhn17 commented:
@dav3r , 

I removed the reference to `vw_cidrs` in the PR description.

dav3r reviewed (approved):
LGTM 👍
